### PR TITLE
Add EADDRNOTAVAIL error to rust

### PR DESCRIPTION
**Pull request by kloenk**
This will be used for network drivers. (#439)

### CODE REVIEW

File: rust/kernel/error.rs [hunk 1]
```diff
@@ -126,6 +126,9 @@ impl Error {
     /// Math result not representable.
     pub const ERANGE: Self = Error(-(bindings::ERANGE as i32));
 
+    /// Cannot assign requested address
```
Comment (ojeda) [inline rust/kernel/error.rs]:
```suggestion
    /// Cannot assign requested address.
```

**Review by ojeda — CHANGES_REQUESTED** [review]

**Comment by ojeda** [comment]
Perhaps we should add all the ones from `include/uapi/asm-generic/errno.h`.

File: rust/kernel/error.rs (hunk 1 above)
Comment (kloenk) [reply rust/kernel/error.rs]: Will fix after the recording

**Comment by kloenk** [comment]
> Perhaps we should add all the ones from `include/uapi/asm-generic/errno.h`.

That would probably be a good idea. But doing it from hand is not the best work to do :-).

Maybe we can build some Marco to do it?

**Comment by kloenk** [comment]
What about a short macro to make it easier, eg that you have to write it like this to declare `ENOEXEC`:
~~~rs
/// Exec format error.
declare_err!(ENOEXEC);

**Comment by bjorn3** [comment]
I think the doc comment will need to be inside the macro invocation for this to work.

**Comment by kloenk** [comment]
really? if the macro just expands to `pub const ENOEXEC: Self = Error(-(bindings::ENOEXEC as i32));`?

**Comment by kloenk** [comment]
I think I will just try it out

**Comment by kloenk** [comment]
Yes, sadly you are right, I think it's still a bit easier to write `declare_err!("Operation not permitted.", EPERM);` instead of `pub const EPERM: Self = Error(-(bindings::EPERM as i32));`

**Comment by kloenk** [comment]
rebased and moved to the macro, Will add missing errors now

**Comment by kloenk** [comment]
@ojeda I now included all errors from `include/uapi/asm-generic/errno.h` as requested.

**Pull request merged**